src/service/jobRoleService.ts

```diff
@@ -12,5 +12,16 @@ export const jobRoleService = {
         catch (e) {
             throw new Error("Failed to get job roles")
         }
+    },
+    async editJobRoles(jobRole: JobRoleToUpdate, jobRoleID: number): Promise<JobRole>{
+        try {
+            const response = await axios.put(`${process.env.API_URL}api/job-roles/${jobRoleID}`, jobRole);
+            if (response.status === 200) {
+                return response.data;
+            }
+            throw new Error('Update failed');
+        } catch (e) {
+            throw new Error('Failed to update job role');
```

Comment:
You will lose any specific error messaging around validation from the API here, so it would be worth doing different things for different status codes